cli/azd/cmd/env.go

```diff
@@ -286,7 +280,7 @@ func (en *envNewAction) Run(ctx context.Context) (*actions.ActionResult, error)
 		subscription:    en.flags.subscription,
 		location:        en.flags.location,
 	}
-	if _, _, err := createAndInitEnvironment(ctx, &envSpec, en.azdCtx, en.console); err != nil {
+	if _, _, err := createAndInitEnvironment(ctx, &envSpec, en.azdCtx, en.console, en.azCli); err != nil {
```

Comment:
For readers that look at these util methods and find that it's dependent on a lot of things to be constructed, which seems cumbersome and backwards. That is indeed true, and we would push towards an object-model where we would construct an environment manager/service up-front, and then the instance method would simply accept scoped variables.

Reply:
100% agreement here - hope to clean this up in future PRs (the next set of changes I want to attack is to clean up createAndInitEnvironment and friends to do less work and as part of that PR, I plan to try to introduce some objects that encapsulate this state.